test-programs/compressed-token-test/tests/test.rs

```diff
@@ -243,27 +248,424 @@ async fn test_mint_to<const MINTS: usize, const ITER: usize>() {
+            mint: mint_1.clone(),
+            token_pool_pda: mint_pool_1,
+            token_program: anchor_spl::token::ID,
+            light_system_program: light_system_program::ID,
+            registered_program_pda: light_system_program::utils::get_registered_program_pda(
+                &light_system_program::ID,
+            ),
+            noop_program: invalid_noop_program.pubkey(),
+            account_compression_authority: light_system_program::utils::get_cpi_authority_pda(
+                &light_system_program::ID,
+            ),
+            account_compression_program: account_compression::ID,
+            merkle_tree: merkle_tree_pubkey,
+            self_program: light_compressed_token::ID,
+            system_program: system_program::ID,
+        };
+        let instruction = Instruction {
+            program_id: light_compressed_token::ID,
+            accounts: accounts.to_account_metas(Some(true)),
+            data: instruction_data.data(),
+        };
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            light_system_program::errors::SystemProgramError::InvalidNoopPubkey.into(),
+        )
+        .unwrap();
+    }
+    // 8. Invalid account compression authority.
+    {
+        let invalid_account_compression_authority = Keypair::new();
+        let accounts = light_compressed_token::accounts::MintToInstruction {
+            fee_payer: payer_1.pubkey(),
+            authority: payer_1.pubkey(),
+            cpi_authority_pda: get_cpi_authority_pda().0,
+            mint: mint_1.clone(),
+            token_pool_pda: mint_pool_1,
+            token_program: anchor_spl::token::ID,
+            light_system_program: light_system_program::ID,
+            registered_program_pda: light_system_program::utils::get_registered_program_pda(
+                &light_system_program::ID,
+            ),
+            noop_program: Pubkey::new_from_array(
+                account_compression::utils::constants::NOOP_PUBKEY,
+            ),
+            account_compression_authority: invalid_account_compression_authority.pubkey(),
+            account_compression_program: account_compression::ID,
+            merkle_tree: merkle_tree_pubkey,
+            self_program: light_compressed_token::ID,
+            system_program: system_program::ID,
+        };
+        let instruction = Instruction {
+            program_id: light_compressed_token::ID,
+            accounts: accounts.to_account_metas(Some(true)),
+            data: instruction_data.data(),
+        };
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            anchor_lang::error::ErrorCode::ConstraintSeeds.into(),
```

Comment:
can we add a specific error here as well?

Reply:
Unfortunately, I think we can't. The constraint looks like:

```
    #[account(seeds = [CPI_AUTHORITY_PDA_SEED], bump)]
    pub cpi_authority_pda: UncheckedAccount<'info>,
```

`seeds` doesn't support adding custom errors with `@`. Quite dumb... After searching through Anchor code, the `@` keyword is supported for:

- `constraint` (e.g. `#[account(constraint = *my_account.key == ID @ MyError::HelloCustom)]`)
- `signer` (e.g. `#[account(signer @ MyError::MyErrorCode)]`)
- `mut` (e.g. `#[account(mut @ MyError::MyErrorCode)]`)
- `address` (e.g. `#[account(address = crate::ID @ MyError::MyErrorCode)]`)

Another TODO Anchor contribution whenever we find time? :stuck_out_tongue: 

We could try to rewrite `seed` checks manually with a bigger `constrant` predicate (not 100% sure whether it's going to work, I can try). I have mixed feelings about that.